packages/lib/src/interactions/RatioSelectionRect.tsx

```diff
@@ -3,22 +3,28 @@ import type { Vector2 } from 'three';
 
 import { useAxisSystemContext } from '../vis/shared/AxisSystemContext';
 import SelectionRect from './SelectionRect';
-import { clampRectangleToVis, getRatioEndPoint } from './utils';
+import { clampRectangleToVis, getRatioRespectingRectangle } from './utils';
 
 interface Props extends SVGProps<SVGRectElement> {
   startPoint: Vector2;
   endPoint: Vector2;
   ratio: number;
+  clampCenter?: boolean;
 }
 
 function RatioSelectionRect(props: Props) {
-  const { startPoint, endPoint, ratio, ...svgProps } = props;
+  const { startPoint, endPoint, ratio, clampCenter, ...svgProps } = props;
 
   const { dataToWorld, worldToData, visSize } = useAxisSystemContext();
 
-  const ratioEndPoint = getRatioEndPoint(startPoint, endPoint, ratio);
+  const [ratioStartPoint, ratioEndPoint] = getRatioRespectingRectangle(
+    startPoint,
+    endPoint,
+    ratio,
+    clampCenter
+  );
```

Comment:
If the center stays at the same place, it means the `startPoint` must move. This means that the code I wrote by assuming `endPoint` is the only vector to move must be refactored.

In the case of the ratio, this is not too difficult: I change the function returning the modified `endPoint` by a function that can modify `startPoint` and `endPoint`. `startPoint` is only modified if `clampCenter` is set to `true`.